Contract API

Krittapas2004 opened the pull request:
I have completed creating the API data for the contract page and added a new controller page with POST, PUT, GET, and DELETE methods for use in Postman. Additionally, I routed all the tables to render server to test if the API is working correctly.

I also have migrate a new data table table using 
`npx prisma migrate dev`

Krittapas2004 committed:
feat(Contract API): Have added code into contract.js in controller

I have added a post,put,get,delete feature for contract api in controller, and export it onto routes contract.js page

Krittapas2004 committed:
feat(Contract API): Import contract api from controller

I have imported the code from the controllers code inside contract.js into routes folder and then export it into app.js

Krittapas2004 committed:
adding env file and removing .env in gitignore

Krittapas2004 committed:
adding the code used in render to create the API in app.js

Krittapas2004 committed:
feat(contract API): creating database table in schema

I have created a database table top store all the information for display on the contract page, including fields for type, resource, location, deadline, and description

Krittapas2004 committed:
Migrating the contract database into schema.prisma file

Navheen0508 reviewed (changes requested):
There's a similar issue with the contracts section - it's saying that the property of create couldn't be found. 

![image](https://github.com/BIT-Studio-3/2024-rest-api-dark-mode/assets/131924590/9d9187fb-531e-4d17-aacf-8176092f9a9a)

I've looked through the code again and found no issues with the way that the controllers, routes or app.js files are set up. However this issue must be resolved for the pull request to be able to be merged into the main as KRUD for contracts is currently not working.

Krittapas2004 committed:
fix(Contract API): Fixed API Data

The issue mirrored that of the agent API; all that was required was migrating a new database onto it. Additionally, I require more information on the database to enable its display in the drop-down box.

Krittapas2004 commented:
### Fixed Error
I've addressed the issue in the code; it mirrored the problem encountered with the agent table during the migration of a new database. Additionally, I've made adjustments to the database model.

Navheen0508 committed:
Merge branch 'staging' into contract-api
